var/spack/repos/builtin/packages/intel-oneapi-cpp/package.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+releases = {'2021.1.2':
+            {'irc_id': '17513', 'build': '63'}}
+
+
+class IntelOneapiCpp(IntelOneApiCompilerPackage):
+    """Intel oneAPI C++ compilers.
+
+    Contains icc, icpc, icx, icpx, dpcpp.
+
+    """
+
+    maintainers = ['rscohn2', 'frankwillmore']
+
+    homepage = 'https://software.intel.com/content/www/us/en/develop/tools/oneapi/components/dpc-compiler.html'
+
+    version('2021.1.2', sha256='68d6cb638091990e578e358131c859f3bbbbfbf975c581fd0b4b4d36476d6f0a', expand=False)
+
+    variant('compiler', values=('icc', 'icx', 'dpcpp'), default='icc', description='Compiler family to use for CC/CXX')
+
+    def __init__(self, spec):
+        self.component_info(
+            dir_name='compiler',
+            components='all',
+            releases=releases,
+            url_name='dpcpp-cpp-compiler')
+        super(IntelOneapiCpp, self).__init__(spec)
+
+    def setup_run_environment(self, env):
```

Comment:
@scheibelp @frankwillmore : Are you OK with deleting all the code that sets CC, CXX, ...

Reply:
It's not clear, e.g. which of the three C++ compilers CXX should point to, and since it's not being set in vars.sh, it's probably best not to set those.

Reply:
I will remove it.

If we want to bring it back at some point in the future, I can make the options be: none,icc,icx and have it default to none. The environment variable setting would only happen if explicitly requested.

Reply:
Will this also drop the variant? I think generally we try to avoid variants which only control post-install behavior, otherwise you might install the same package multiple times just to choose whether to expose icc/icx. 

If users need to configure how to set CC etc. for builds outside of Spack, then this could be useful. Spack could also add support for per-package environment configuration (https://spack-tutorial.readthedocs.io/en/latest/tutorial_modules.html#add-custom-environment-modifications shows how this can be done for generated module files but not for `spack load` for example).